src/lib/comms/sol-socket-dtls.h

```diff
@@ -37,7 +37,7 @@ enum sol_socket_dtls_cipher {
 struct sol_socket_dtls_credential_cb {
     const void *data;
 
-    void *(*init)(const void *data);
+    int (*init)(const void *data);
```

Comment:
better take the opportunity to fix the signature to include a size, or `struct sol_str_slice` (slices can contain binary data, including \0, as they specify the size).